caffe2/operators/distance_op.cc

```diff
@@ -223,10 +223,27 @@ bool DotProductOp<float, CPUContext>::RunOnDevice() {
   return true;
 }
 
+vector<TensorShape> TensorInferenceForDotProduct(
+    const OperatorDef& /* def */,
+    const vector<TensorShape>& in) {
+  vector<TIndex> dims;
+  if (in[0].dims().size() > 0) {
```

Comment:
this will fail if 'in' is empty. Do you mind guarding this via an ENFORCE ?

Reply:
Ah, sure I can put the guard there.